contracts/pcv/balancer/BalancerPCVDepositWeightedPool.sol

```diff
@@ -0,0 +1,323 @@
+        return address(token);
+    }
+
+    // @notice deposit tokens to the Balancer pool
+    function deposit() external override whenNotPaused {
+        uint256[] memory balances = new uint256[](poolAssets.length);
+        uint256 totalbalance = 0;
+        for (uint256 i = 0; i < balances.length; i++) {
+            balances[i] = IERC20(address(poolAssets[i])).balanceOf(address(this));
+            totalbalance += balances[i];
+        }
+        require(totalbalance > 0, "BalancerPCVDepositWeightedPool: no tokens to deposit");
+
+        // Read oracles
+        uint256[] memory underlyingPrices = _readOracles();
+
+        // Build joinPool request
+        if (feiInPool) {
+            // If FEI is in pool, we mint the good balance of FEI to go with the tokens
+            // we are depositing
+            uint256 _feiToMint = underlyingPrices[tokenIndexInPool] * balances[tokenIndexInPool] / 1e18;
+            _mintFei(address(this), _feiToMint);
+            balances[feiIndexInPool] = _feiToMint;
+        }
+        bytes memory userData = abi.encode(IWeightedPool.JoinKind.EXACT_TOKENS_IN_FOR_BPT_OUT, balances, 0);
+
+        IVault.JoinPoolRequest memory request = IVault.JoinPoolRequest({
+            assets: poolAssets,
+            maxAmountsIn: balances,
+            userData: userData,
+            fromInternalBalance: false // tokens are held on this contract
+        });
+
+        // approve spending on balancer's vault
+        for (uint256 i = 0; i < balances.length; i++) {
+            if (balances[i] > 0) {
+                IERC20(address(poolAssets[i])).approve(address(vault), balances[i]);
+            }
+        }
+
+        // execute joinPool & transfer tokens to Balancer
+        uint256 bptBalanceBefore = IWeightedPool(poolAddress).balanceOf(address(this));
+        vault.joinPool(
+            poolId, // poolId
+            address(this), // sender
+            address(this), // recipient
+            request // join pool request
+        );
+        uint256 bptBalanceAfter = IWeightedPool(poolAddress).balanceOf(address(this));
+
+        // Check for slippage
+        {
+            // Compute USD value deposited
+            uint256 valueIn = 0;
+            for (uint256 i = 0; i < balances.length; i++) {
+                valueIn += balances[i] * underlyingPrices[i] / 1e18;
+            }
+
+            // Compute USD value out
+            uint256 bptPrice = _getBPTPrice(underlyingPrices);
+            uint256 valueOut = Decimal.from(bptPrice).mul(bptBalanceAfter - bptBalanceBefore).div(1e18).asUint256();
+            uint256 minValueOut = Decimal.from(valueIn)
+                .mul(Constants.BASIS_POINTS_GRANULARITY - maximumSlippageBasisPoints)
+                .div(Constants.BASIS_POINTS_GRANULARITY)
+                .asUint256();
+            require(valueOut > minValueOut, "BalancerPCVDepositWeightedPool: slippage too high");
+        }
```

Comment:
This will cost a lot of gas. I'd strongly prefer using built-in slippage checks here

Reply:
how to compute the minimum amount of bpt out, though ?